components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConstants.java

```diff
@@ -773,6 +773,11 @@ private Permissions() {
     public static final String API_RESTAPI_ETAG_SKIP_URI_HTTPMETHOD = API_RESTAPI_ETAG_SKIP_URI + "HTTPMethods";
 
     public static final String JWT_EXPIRY_TIME = API_KEY_VALIDATOR + "JWTExpiryTime";
+    public static final String JWT_AUDIENCES = "JWTAudiences";
+    public static final String JWT_AUDIENCE = "JWTAudience";
+    public static final String AUDIENCE = "Audience";
+    public static final String BASEPATH = "Basepath";
+    public static final String URN_CHOREO = "urn:choreo:";
```

Comment:
Shall we remove choreo specific stuff from the code? Ideally this should be configurable right?

Reply:
Actually `"urn:choreo:"` is used to modify the incoming scopes list into a form which is acceptable by the APIM. What we recieve in the JWT will be `urn:choreo:org-id-1:apim:api_create` and what we required for scope validation is only the `apim:api_create`. So we use this to remove `APIConstants.URN_CHOREO + orgId + ":"` form the scopes. Since scope validation flow is common for both opaque and JWT, we have generalised the validation process.